pkg/middleware/logging.go

```diff
@@ -0,0 +1,37 @@
+// Copyright KubeArchive Authors
```

Comment:
What is this package for?

Reply:
`gin` does not support `slog`, so this replaces their `Logger` middleware by our own, which uses `slog`.